doc/build/sysbuild/index.rst

```diff
@@ -0,0 +1,375 @@
+build.
+
+Using CMake and ninja:
+
+.. zephyr-app-commands::
+   :tool: cmake
+   :app: sysbuild
+   :board: reel_board
+   :goals: build
+   :gen-args: -DAPP_DIR=<abs-path>/samples/hello_world
+   :compact:
+
+.. note::
+   When using ``west build`` a relative path to the application can be used, but
+   if using CMake and ``APP_DIR`` directly then it is important to specify the
+   absolute path to the sample.
+
+
+Configuration namespacing
+*************************
+
+When build a single Zephyr application, all CMake cache settings and Kconfig
+build options given on the command line as ``-D<var>=<value>`` or
+``-DCONFIG_<var>=<value>`` are handled by the single Zephyr CMake build system.
+
+Sysbuild may combine multiple Zephyr applications as providing a Kconfig tree
+for sysbuild configuration.
+
+Therefore there is a need for configuration namespacing so that users can
+specify if a setting controls sysbuild or a Zephyr application.
+
+.. _sysbuild_cmake_namespace:
+
+CMake variable namespacing
+**************************
+
+CMake variables are all variables that are passed to CMake using
+``-D<var>=<value>``, except settings intended for Kconfigs which takes the form
+``-DCONFIG_<var>=<value>`` or ``-D<namespace>_CONFIG_<var>=<value>``.
+
+Sysbuild is the entry point for the build system, and therefore all CMake
+variables are first processed by sysbuild.
+
+Each image will have a namespace defined by its application name, see
+:ref:`sysbuild_zephyr_application`.
+
+The setting that must be passed to sysbuild for an application must be prefixed
+with the namespace, as::
+
+  -D<namespace>_<var>=<value>
+
+For example passing of ``FOO`` setting to ``my_sample`` application is done as::
+
+  cmake -Dmy_sample_FOO=BAR ...
+
+
+Kconfig namespacing
+*******************
+
+Kconfig settings are variables that takes the form ``-DCONFIG_<var>=<value>``
+or ``-D<namespace>_CONFIG_<var>=<value>``.
+
+Kconfig settings controlling the sysbuild Kconfig tree are identified by
+``SB_CONFIG``, so a setting given as ``-DSB_CONFIG_<var>=<value>`` will be used
+by the sysbuild Kconfig tree.
+
+The Kconfig tree for each image will have a namespace defined by its application
+name, see :ref:`sysbuild_zephyr_application`.
+
+The Kconfig setting that must be given to sysbuild in order to be directed to an
+application must be prefixed with the namespace, as::
+
+  -D<namespace>_CONFIG_<var>=<value>
+
+For example passing of Kconfig ``FOO`` setting to ``my_sample`` Kconfig tree is
+done as::
+
+  cmake -Dmy_sample_CONFIG_FOO=BAR ...
+
+Kconfig settings given as ``-DCONFIG_<var>=<value>`` is passed to the main
+application.
+This makes passing of Kconfig settings to main application identical between
+building with or without using sysbuild.
+If ``my_sample`` is the main application, then Kconfig settings can be given
+both as ``-DCONFIG_<var>=<value>`` or ``-Dmysampl_CONFIG_<var>=<value>``.
```

Comment:
Typo for sample, and should be `-Dmy_sample_CONFIG...` with underscore

Reply:
done